src/main/ruby/truffleruby/core/main.rb

```diff
@@ -85,3 +85,7 @@ def to_s
     end
   end
 end
+
+Truffle::Boot.delay do
+  Random::DEFAULT = Random.new
+end
```

Comment:
Any reason to move it here instead of having the `delay` block in `core/random.rb`?

Reply:
Part of the idea of these `delay` blocks is to keep the code together as much as possible.

Reply:
It causes a long error (starts failing at failing to load the core library) since something that's needed there is not yet loaded in `random.rb`

Reply:
That's unexpected, I'll try to reproduce.